wrappers/services.go

```diff
@@ -277,10 +277,27 @@ func AddSnapServices(s *snap.Info, inter interacter) (err error) {
 		}
 
 		svcName := app.ServiceName()
-		if err := sysd.Enable(svcName); err != nil {
-			return err
+
+		// if this service is in the list of services to be left disabled, don't
+		// enable it
+		shouldEnable := true
+		for _, disabledSvc := range disabledSvcs {
+			// sanity check that all apps in the disabledSvcs list are in fact
+			// services
+			if !disabledSvc.IsService() {
+				return fmt.Errorf("snap app %s is not a service, cannot disable", disabledSvc.Name)
+			}
+			if disabledSvc.Name == app.Name {
+				shouldEnable = false
+			}
+		}
+
+		if shouldEnable {
```

Comment:
then you could do the following here:
```
if strutil.ListContains(disabledSvcs, app.Name) {
    // service is disabled, nothing to do
    continue
}

if err := sysd.Enable(svcName); err != nil {
    return err
}
enabled = append(enabled, svcName)
```

Reply:
yes this is simpler, I ended up going with this, thanks